go/client/logclient.go

```diff
@@ -277,6 +284,37 @@ func (c *LogClient) AddChainWithContext(ctx context.Context, chain []ct.ASN1Cert
 	return c.addChainWithRetry(ctx, AddChainPath, chain)
 }
 
+func (c *LogClient) AddJSON(data []byte) (*ct.SignedCertificateTimestamp, error) {
+	req := addJSONRequest{
+		Data: base64.StdEncoding.EncodeToString(data),
```

Comment:
And `base64.StdEncoding.EncodeToString(data)` for just `data`, it will then take anything, and the JSON encoder already base64-encodes a `byte[]`, so you'll get exactly the same result as now without changing anything to the calling code, while being more general (accepting anything that will encode to JSON).